arrows-stdlib-cats/src/main/scala/arrows/stdlib/cats.scala

```diff
@@ -0,0 +1,114 @@
+package arrows.stdlib
+
+import cats.{ Applicative, StackSafeMonad }
+import cats.effect._
+import cats.arrow.ArrowChoice
+import cats.kernel.{ Monoid, Semigroup }
+import cats.mtl.ApplicativeAsk
+
+import scala.concurrent.{ ExecutionContext, Promise }
+import scala.util.control.NonFatal
+import scala.util.{ Failure, Success }
+
+object Cats extends ArrowInstances
+
+sealed abstract class ArrowInstances extends ArrowInstances1 {
+
+  implicit def catsEffectForTask(implicit ec: ExecutionContext): Effect[Task] = new ArrowAsync[Unit] with Effect[Task] {
+    def runAsync[A](fa: Task[A])(cb: Either[Throwable, A] => IO[Unit]): SyncIO[Unit] =
+      SyncIO { fa.run(())(ec); () }
+  }
+
+  implicit val catsArrowChoiceForArrow: ArrowChoice[Arrow] = new ArrowChoice[Arrow] {
+    def choose[A, B, C, D](f: Arrow[A, C])(g: Arrow[B, D]): Arrow[Either[A, B], Either[C, D]] =
+      Arrow[Either[A, B]].flatMap {
+        case Left(a)  => f(a).map(Left(_))
```

Comment:
The bytecode for this transformation will be large. What do you think about optimizing it? Something like

```scala
// In a stable scope
private final object ToLeft {
   private final val instance: Any => Left[Any] = Left(_)
   def apply[T]() = instance.asInstanceOf[T => Left[T]]
}

private final object ToRight {
   private final val instance: Any => Right[Any] = Right(_)
   def apply[T]() = instance.asInstanceOf[T => Right[T]]
}
```

and then

```scala
Arrow[Either[A, B]].flatMap { a =>
  if(a.isLeft) f(a).map(toLeft)
  else f(a).map(toRight)
}
```